### PR TITLE
MM-54778 Fix mark as unread on GMs

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
When we mark as unread the GMs, we don't get the number badges, unlike DMs.

This is because we missed a change on the "GM as DM" PR, in particular, the part that counts the number of mentions in a channel after marking it as unread.

Considering the groups messages there is enough.

#### Ticket Link
Fix https://mattermost.atlassian.net/browse/MM-54778

#### Release Note
If it reaches the same version as the "GM as DM" PR, no release note needed
```release-note
Fix issue where marking as unread a GM won't show the mention badges correctly.
```

### CODE REVIEW

**Review by hmhealey — APPROVED** [review]

**Review by harshilsharma63 — APPROVED** [review]

**Comment by harshilsharma63** [comment]
No tests needed?

**Review by yasserfaraazkhan — APPROVED** [review]
after marking as unread, the badge shows correct numbered notification badge.

**Comment by larkox** [comment]
Re-requesting review since I have changed the behavior of the way we get the mentions, and may have bigger implications.

File: server/channels/store/sqlstore/channel_store.go [hunk 1]
```diff
@@ -2695,7 +2695,7 @@ func (s SqlChannelStore) CountPostsAfter(channelId string, timestamp int64, user
 		})
 
 	if userId != "" {
-		query = query.Where(sq.Eq{"UserId": userId})
```
Comment (hmhealey) [inline server/channels/store/sqlstore/channel_store.go:2698]: Would you mind renaming this parameter to something like `excludeUserId` that makes it clearer that this works opposite to how it did before?

File: server/channels/store/sqlstore/channel_store.go [hunk 2]
```diff
@@ -2657,7 +2657,7 @@ func (s SqlChannelStore) CountUrgentPostsAfter(channelId string, timestamp int64
 		})
 
 	if userId != "" {
-		query = query.Where(sq.Eq{"Posts.UserId": userId})
```
Comment (hmhealey) [inline server/channels/store/sqlstore/channel_store.go:2660]: Same here

**Review by hmhealey — CHANGES_REQUESTED** [review]

**Review by harshilsharma63 — APPROVED** [review]

**Comment by larkox** [comment]
@agarciamontoro Adding you as reviewer since I am changing how some things work on the server side, and just verify I am not missing anything in terms of performance and such.

This is a "rare" action (mark a channel as unread) so there should be not many performance implications, but better safe than sorry.

**Review by hmhealey — APPROVED** [review]

**Review by agarciamontoro — APPROVED** [review]
Changes look good, thanks!

First a nit: The same change `userID` > `excludedUserID` you did in the declaration of `CountPostsAfter` should also be done in the definition of `CountPostsAfter` and in the declaration and definition of `CountUrgentPostsAfter`.

Now on the performance side of things: I don't foresee any impact from this change, given that the query is triggered by a not so common user action, and usually with a fairly small number of posts. It is safe to merge without a load-test.

**Comment by larkox** [comment]
@yasserfaraazkhan Make sure when testing that your own posts do not contribute to the badge number.

Also, make sure the DMs and Urgent feature also work as it should.

**Comment by larkox** [comment]
@yasserfaraazkhan Finally the server is updated and the checks are green, so... this is ready to test :smile:

**Review by yasserfaraazkhan — APPROVED** [review]
thank you @larkox the badge does not include logged in users posts.

Screenshot 2023-10-24 at 6 53 01 PM
 

Screenshot 2023-10-24 at 6 53 06 PM

image

**Pull request merged**